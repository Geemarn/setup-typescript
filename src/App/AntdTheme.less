// import default antd less file NB: can be ovv=erride by craco config file
@import "~antd/dist/antd.less";



