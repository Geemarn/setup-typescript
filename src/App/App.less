@import "AntdTheme";

body {
  padding: 0;
  @media screen and (max-width: 800px) {
    padding: 10px;
  }
}
