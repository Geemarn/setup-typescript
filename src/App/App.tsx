import React, { lazy } from "react";
import { ConfigProvider } from "antd";
import { useSelector } from "react-redux";
import { Route, Switch } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { theme } from "../_shared/theme/themeVariables";
import { ReducerState } from "../redux/reducers/types";
import "./App.less";

const Layout = lazy(() => import("../layout"));

function App(props: any) {
  //bring in UI layout from redux state
  const { darkMode, rtl, topMenu } = useSelector((state: ReducerState) => ({
    darkMode: state.ui.switchTheme,
    rtl: state.ui.switchRTL,
    topMenu: state.ui.topMenu,
  }));

  return (
    <ConfigProvider direction={rtl ? "rtl" : "ltr"}>
      <ThemeProvider theme={{ ...theme, rtl, topMenu, darkMode }}>
        <Switch>
          <Route exact path="/" component={Layout} />
        </Switch>
      </ThemeProvider>
    </ConfigProvider>
  );
}

export default App;
