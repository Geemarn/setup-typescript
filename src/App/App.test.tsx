import React from 'react';
import { shallow, mount, render } from 'enzyme';
// import { render, screen } from '@testing-library/react';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it('expect to render app component', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
