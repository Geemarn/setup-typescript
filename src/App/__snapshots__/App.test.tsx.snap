// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render app component 1`] = `
<div
  className="app"
>
  <WingBlank
    prefixCls="am-wingblank"
    size="lg"
  >
    <h1>
      this is a react bootstrap projects
    </h1>
    <Button
      activeStyle={Object {}}
      disabled={false}
      inline={false}
      loading={false}
      prefixCls="am-button"
      size="large"
      type="warning"
    >
      Start
    </Button>
    <WhiteSpace
      prefixCls="am-whitespace"
      size="md"
    />
    <Button
      activeStyle={Object {}}
      disabled={false}
      inline={false}
      loading={false}
      prefixCls="am-button"
      size="large"
      type="primary"
    >
      Start
    </Button>
    <br />
    <Button
      type="primary"
    >
      Start here
    </Button>
  </WingBlank>
</div>
`;
